src/modules/auth/redux/actions.ts

```diff
@@ -18,3 +19,129 @@ export const authActions = {
     );
   },
 };
+
+export const setCurrentUser = (currentUser) => ({
+  type: AuthConsts.SET_CURRENT_USER,
+  payload: {
+    currentUser 
+  }, 
+})
+
+export const getCurrentUser = () => async (dispatch: Dispatch) => {
+  try {
+    const currentUser = await AuthService.getCurrentUser();
+
+    dispatch(setCurrentUser(currentUser));
+  }catch(error) {
+    console.error(error);
+    message.error('Unkown Error', error);
+  }
+}
+
+export const setAuthState = (state) => ({
+  type: AuthConsts.SET_AUTH_STATE,
+  payload: {
+    state, 
+  }
+})
+
+export const setAuthData = (data) => ({
+  type: AuthConsts.SET_AUTH_DATA,
+  payload: {
+    data, 
+  }
+})
+
+const resolveAuthFromStorage = () => {
+  const localSession = localStorage.getItem(SESSION_KEY) || '';
+  
+  try {
+    const sessionObj = JSON.parse(localSession);
+
+    return sessionObj;
+  }catch(error) {
+    console.error(error)
+
+    return {}
+  }
+}
+
+export const resolveAuthState = () => async(dispatch: Dispatch) => {
+  dispatch(setAuthState(AuthState.Pending));
+
+  const session = resolveAuthFromStorage();
+  console.log("data", session)
+  if(session.access_token) {
+    dispatch(setAuthData(session));
+    dispatch(setAuthState(AuthState.Authorized));
+  }
+  else {
+    dispatch(setAuthState(AuthState.Unauthorized));
+  }
+}
+
+export const resolveJWTAuthState = () => async(dispatch: Dispatch) => {
+  dispatch(setAuthState(AuthState.Pending));
+
+  const session = resolveAuthFromStorage();
+  
+  if(session.access_token) {
+    dispatch(refreshToken(session.access_token, session.email));
+  }
+  else {
+    dispatch(setAuthState(AuthState.Unauthorized));
+  }
+}
+
+export const login = (form: AuthModel.SigninPayload) => async(dispatch: Dispatch) => {
+  dispatch(setAuthState(AuthState.Pending));
```

Comment:
Cause we use redux to control auth state, component can use `useSelect` to handle the authState changed.
Maybe we don't need to inject dispatch?

Reply:
This is not manually injection, but taking adv. of redux-thunk.

Reply:
get it